aten/src/ATen/native/native_functions.yaml

```diff
@@ -2916,6 +2921,9 @@
   dispatch:
     DefaultBackend: permute
 
+- func: cov(Tensor self, Tensor? other=None, Tensor? fweights=None, Tensor? aweights=None, bool? bias=None, int? ddof=None) -> Tensor
```

Comment:
Let's make sure this has the same signature as the NumPy function. Not all parameters have to be supported, and putting values in for them can throw an error explaining the functionality is not implemented, but the signatures should still match.

Here the first two tensors look good, but rowvar, bias, ddof need to appear in that order before fweights. For the optional parameters the corresponding cpp signature must accept an optional, too.

Reply:
I tried these things related to cpp definition and hacky wrapper.

1) When I remove the hacky wrapper and try to make the arguments as optional in the cpp signature it throws a rather strange error
```
/usr/bin/ld: /home/rohan/codes/Dev/pytorch/build/lib/libtorch_cpu.so: undefined reference to `at::native::cov(at::Tensor const&, at::Tensor const&, at::Tensor const&, at::Tensor const&, c10::optional<bool>, c10::optional<long>)'
collect2: error: ld returned 1 exit status
```
while my cpp signature changed to 
```
Tensor cov(const Tensor& self, c10::optional<Tensor>& otherOpt,
           c10::optional<Tensor>& fweightsOpt, 
           c10::optional<Tensor>& aweightsOpt, 
           c10::optional<bool> biasOpt, c10::optional<int64_t> ddofOpt)
```
keeping native_functions definition the same.
From what I see most (around 90%) of the functions which use multiple optional tensors have used hack wrapper and I couldnt find another function which dealt with optional tensors directly. Any idea what I could try?

Reply:
issues related to optional tensors were recently fixed in #50767 - are you based after it? Try rebasing to the latest, let us know if the issue persists.

Reply:
@ngimel if I understand correctly I have to rebase to the branch to which the PR was merged into ? I am a little confused as to what exactly I have to rebase onto to reflect the changes in that particular PR. Could you please help me out?

Reply:
If you rebase to the current master, you should be fine.